packages/lib-classifier/src/plugins/drawingTools/models/marks/Mark/Mark.js

```diff
@@ -13,6 +13,9 @@ const BaseMark = types.model('BaseMark', {
     TextTask.AnnotationModel
   )),
   frame: types.optional(types.number, 0),
+  subTaskMarkBounds: types.optional(types.frozen({}), undefined),
+  subTaskPreviousAnnotations: types.map(TextTask.PreviousAnnotationsModel),
+  subTaskVisibility: types.optional(types.boolean, false),
```

Comment:
Should these be volatile, so that they aren't included in the classification when the mark is serialised?

If they aren't needed globally in the page, or in the classification, then they can be stored in component state for each mark. I'm not 100% sure if that would work.

Reply:
Oh, volatile would be a much better pick. Thanks for the suggestion. 

I think the previous annotations and visibility needs to be in the MST store state. Previous annotations need to be stored and available for the `TranscribedLines` and `SubTaskPopup` components. The visibility ensures only one subtask can be open at a time. The bounds, however, may be a candidate for local component state.

Reply:
If the popup uses `React.Portal`, then you could wrap it inside its parent line and pass the line bounds down as a prop.
```jsx
<Line>
  <SubtaskPopup/>
</Line>
```

That might be too big of a change for this PR. Portals are really useful when you want to pass data to another part of the page.

Reply:
If only one subtask can be active at a time then we could rework that, at some point, so that it works similarly to active marks ie. store the active subtask as an optional property of the parent Task. I think that would avoid having to track a bunch of booleans, and enforce that only one is true at any time by only ever storing one at a time.

Reply:
All great ideas, but yeah, I agree that it's too much for this PR. Could you open an issue tracking these suggestions?